Options.py

```diff
@@ -1146,22 +1146,26 @@ class OptionGroup(typing.NamedTuple):
 def get_option_groups(world: typing.Type[World], visibility_level: Visibility = Visibility.template) -> typing.Dict[
         str, typing.Dict[str, typing.Type[Option[typing.Any]]]]:
     """Generates and returns a dictionary for the option groups of a specified world."""
-    option_groups = {option: option_group.name
-                     for option_group in world.web.option_groups
-                     for option in option_group.options}
+    option_to_name = {option: option_name for option_name, option in world.options_dataclass.type_hints.items()}
```

Comment:
This looks like it's relying on the assumption that one `Option` class is not used multiple times in a game's options, which is not strictly true, but it's probably ok.

Do we have a unit test that checks to make sure that `Option` classes aren't used more than once unless they are a subclass of `Removed` (or that they all have `Visibility.none`)?
because this can fail otherwise.

Reply:
Can confirm that there's no unit test or errors from using the same class twice and so this would only include the last one

Reply:
#3530